### PR TITLE
[Enhancement] add build paramnt ENABLE_JAVAUDF, BUILD_DOCS

### LINKED ISSUE #8611: [Enhancement] add build paramnt ENABLE_JAVAUDF, BUILD_DOCS
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

add build paramnt ENABLE_JAVAUDF, BUILD_DOCS

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes
@Gabriel39  Please take a look.
Issue Number: close #8611

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
LGTM

File: build.sh [hunk 1]
```diff
@@ -283,27 +291,32 @@ if [ ${BUILD_BE} -eq 1 ] ; then
...
-    if [ ${BUILD_FE} -eq 0 ]; then
-      ${MVN_CMD} package -pl fe-common,fe-core,java-udf -DskipTests
+    if [ ${ENABLE_JAVAUDF} = "ON" ] ; then
+        echo "Build Frontend Modules: java-udf"
+        cd ${DORIS_HOME}/fe
+        if [ ${CLEAN} -eq 1 ]; then
+            clean_fe
+        fi
+        if [ ${BUILD_FE} -eq 0 ]; then
+            ${MVN_CMD} package -pl fe-common,fe-core,java-udf -DskipTests
```
Comment (Gabriel39) [inline build.sh:307]:
```suggestion
            ${MVN_CMD} package -pl ${FE_MODULES} -DskipTests
```

File: build.sh [hunk 2]
```diff
@@ -283,27 +291,32 @@ if [ ${BUILD_BE} -eq 1 ] ; then
...
-      ${MVN_CMD} package -pl fe-common,fe-core,java-udf -DskipTests
+    if [ ${ENABLE_JAVAUDF} = "ON" ] ; then
+        echo "Build Frontend Modules: java-udf"
+        cd ${DORIS_HOME}/fe
+        if [ ${CLEAN} -eq 1 ]; then
+            clean_fe
+        fi
+        if [ ${BUILD_FE} -eq 0 ]; then
+            ${MVN_CMD} package -pl ${FE_MODULES} -DskipTests
+        fi
```
Comment (yangzhg) [inline build.sh:308]: just remove this block ?

**Review by yangzhg — COMMENTED** [review]
I think libjvm.so should not link at compile time. it shoud loaded at java udf executing

**Comment by Gabriel39** [comment]
> I think libjvm.so should not link at compile time. it shoud loaded at java udf executing

@yangzhg I agree. I'll submit next PR to solve this in near future

**Review by yangzhg — DISMISSED** [review]
LGTM

File: build.sh [hunk 3]
```diff
@@ -254,14 +257,14 @@ make
 
 # Assesmble FE modules
 FE_MODULES=
-if [ ${BUILD_FE} -eq 1 -o ${BUILD_SPARK_DPP} -eq 1 -o ${ENABLE_JAVAUDF} = "ON" ]; then
-    if [ ${BUILD_FE} -eq 1 -a ${ENABLE_JAVAUDF} = "ON" ]; then
+if [ ${BUILD_FE} -eq 1 -o ${BUILD_SPARK_DPP} -eq 1 -o ${BUILD_JAVA_UDF} = "ON" ]; then
+    if [ ${BUILD_FE} -eq 1 -a ${BUILD_JAVA_UDF} = "ON" ]; then
```
Comment (yangzhg) [inline build.sh:261]: BUILD_JAVA_UDF may by `0` and `1` not `on` `off`

**Review by yangzhg — APPROVED** [review]
LGTM

**Comment by yangzhg** [comment]
merger it for continue compile and deploy

**Pull request merged**